src/com/google/enterprise/adaptor/database/DatabaseAdaptor.java

```diff
@@ -193,12 +193,20 @@ public void init(AdaptorContext context) throws Exception {
       log.config("adaptor runs in lister-only mode");
     }
 
-    modeOfOperation = cfg.getValue("db.modeOfOperation");
+    modeOfOperation = cfg.getValue("db.modeOfOperation").trim();
     if ("urlAndMetadataLister".equals(modeOfOperation) && encodeDocId) {
       String errmsg = "db.modeOfOperation of \"" + modeOfOperation
           + "\" requires docId.isUrl to be \"true\"";
       throw new InvalidConfigurationException(errmsg);
     }
+    String modeOfOperationValid[] = {"blobColumn", "filepathColumn",
```

Comment:
Please either change this to use the same reflection mechanism that is used to do the work, or at least add a TODO(jlacey) for the same. I do not want to manually keep this list up-to-date. It already conflicts with another CL.